### ISSUE TITLE
New Radiation-Hydro Extension

### ISSUE THREAD

**Issue by ax3l** [ComputationalRadiationPhysics/openPMD-standard#134]
@richardbriggs and @CFGrote are planning an extension for radiation hydro codes (first Multi & Esther; more exist such as: Hyades, Helios, Flash, ...).

This issue documents the progress so far.

openPMD extension name: `Rad-Hydro`

openPMD extension ID: `2` (bitmask!)
### Naming Conventions for Required Records
- temperature
- ionization state / density
- laser intensity
- pressure
- ...
### Required Attributes
- eq. of state (model; can be several): `BLF`, `SESAME`, ..., `other`
  - investigate more
  - link default references for each
- range of validity of eq. of state (density, temperature)
- interpolation scheme _outside_ range of v alidity for eq. of state (linear, b-spline, polynomial, ...)
- type of code: Lagrangian / non-Lagrangian
  - investigate more
  - link default references for each
- properties of code: (non-) energy conserving, ...
  - investigate more
  - link default references for each
- laser profile: `plane wave`, `other`
- temporal laser profile (define `t=0` point with regards to spatial simulation extent!)

**Comment by ax3l** [comment]
@richardbriggs @CFGrote I know Richard is currently between jobs but just a quick question:
Are you already using an [`openPMDextension`](https://github.com/openPMD/openPMD-standard/blob/1.0.0/STANDARD.md#hierarchy-of-the-data-file) ID for your hydro files?

Above I proposed/reserved `2`. Just wanted to know to avoid collisions.

**Comment by CFGrote** [comment]
i have no information. @RichardBriggs ? Would it make sense ( I guess it
would) to find a more self-descriptive enumeration scheme for openpdm
flavours? Is it possible to use strings instead of ints?


On Tue, Nov 21, 2017 at 01:11:53PM +0000, Axel Huebl wrote:
> @richardbriggs @CFGrote I know Richard is currently between jobs but just a quick question:
> Are you already using an [`openPMDextension`](https://github.com/openPMD/openPMD-standard/blob/1.0.0/STANDARD.md#hierarchy-of-the-data-file) ID for your hydro files?
>
> Above I proposed `2`. Just wanted to know to avoid collisions.
>
> --
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/openPMD/openPMD-standard/issues/134#issuecomment-346022183

--
Dr. Carsten Fortmann-Grote
Scientist for Scientific Simulations / Wissenschaftler fuer wissenschaftliche Simulationen
European XFEL GmbH
Holzkoppel 4
22869 Schenefeld
Germany

Phone: +49 (0)40 8998-5603
Fax:   +49 (0)40 8998-1905
Email: carsten.grote@xfel.eu
Web: www.xfel.eu

Managing Directors: Prof. Dr. Robert Feidenhans'l, Dr. Claudia Burger

Registered as European X-Ray Free-Electron Laser Facility GmbH
at Amtsgericht Hamburg, HRB 111165

**Comment by richardbriggs** [comment]
Currently it is set to openPMDextension=2, I don’t have any further information beyond that as I haven’t worked on modifying the opmd files from the hydro files outputs. 

I’ve made a note to spend some time looking at this in my final week (ending Dec 8th). I have 2 weeks of beamtime unntil then :/

Richard

**Comment by ax3l** [comment]
Thank you for the quick answer! No problem, wish you good luck in the next weeks!

**Comment by ax3l** [comment]
> Would it make sense ( I guess it would) to find a more self-descriptive enumeration scheme for openpdm flavours?

yes, absolutely: we are drafting to change that in https://github.com/openPMD/openPMD-standard/issues/151
